pages/index.js

```diff
@@ -22,98 +22,92 @@ const months = [
 ];
 
 const Home = () => {
-    const [talks, setTalks] = React.useState(null);
-    const [nextTalk, setNextTalk] = React.useState(null);
-    const [nextTweet, setNextTweet] = React.useState(null);
+    //todo improve in next PR
+    const {
+        data: { talks },
+        isLoading: isLoadingTalks,
+        isError: isErrorTalks,
+    } = useDataApi('/api/talks', {
+        talks: [],
+    });
 
-    React.useEffect(() => {
-        const fetchTalks = async () => {
-            const { talks, error } = await fetch('/talks');
-            if (error) {
-                return console.error(error);
-            }
-
-            setTalks(talks);
-            setNextTalk(talks.find(talk => new Date(talk.date) > new Date()));
-        };
-
-        fetchTalks();
-    }, []);
-
-    React.useEffect(() => {
-        const fetchTweets = async () => {
-            const { tweets, error } = await fetch('/tweets');
-            if (error) {
-                return console.log(error);
-            }
+    const {
+        data: { tweets },
+        isLoading: isLoadingTweets,
+        isError: isErrorTweets,
+    } = useDataApi('/api/tweets', {
+        tweets: [],
+    });
 
-            setNextTweet(tweets.find(tweet => new Date(tweet.date) > new Date()));
-        };
+    const isLoadingTalksElement = isLoadingTalks || isErrorTalks || !talks.length;
+    const isLoadingTweetsElement = isLoadingTweets || isErrorTweets || !tweets.length;
```

Comment:
There are some edge cases not handled through this shortcut : An error, or an absence of talk/tweet will lead to an indefinite loading state
To fix in an upcoming PR?

Reply:
Reponse above (tldr: create issue about better error managment in front)